### PR TITLE
Reinstate elementtree example

### LINKED ISSUE #4426: elementtree example doesn't work, uses classical mappers
multiple issues with this example:

1. can't be run as python examples/elementtree/optimized_ai.py becuase it has a file named "pickle" which gets in the way of SQLAlchemy "import pickle"

2. run as a module, has a unicode error on both py3k *and* py2k:

```
Traceback (most recent call last):
  File "/opt/python-3.7.0/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/python-3.7.0/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/classic/dev/sqlalchemy/examples/elementtree/optimized_al.py", line 223, in <module>
    print(document)
  File "/home/classic/dev/sqlalchemy/examples/elementtree/optimized_al.py", line 87, in __str__
    self.element.write(buf)
  File "/opt/python-3.7.0/lib/python3.7/xml/etree/ElementTree.py", line 777, in write
    short_empty_elements=short_empty_elements)
  File "/opt/python-3.7.0/lib/python3.7/contextlib.py", line 119, in __exit__
    next(self.gen)
  File "/opt/python-3.7.0/lib/python3.7/xml/etree/ElementTree.py", line 836, in _get_writer
    yield file.write
  File "/opt/python-3.7.0/lib/python3.7/contextlib.py", line 511, in __exit__
    raise exc_details[1]
  File "/opt/python-3.7.0/lib/python3.7/contextlib.py", line 496, in __exit__
    if cb(*exc_details):
  File "/opt/python-3.7.0/lib/python3.7/contextlib.py", line 383, in _exit_wrapper
    callback(*args, **kwds)
TypeError: string argument expected, got 'bytes'

```

3. uses classical mappings which will be going away

4. also uses query.join(..., aliased=True) which would be nice to get rid of, but this is actually a demo of what that flag was intended for.   might be nice though to use aliased() explicitly in any case.

### PR DESCRIPTION
**Pull request by vrajmohan**
Partially fixes sqlalchemy/sqlalchemy#4426.

### CODE REVIEW

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision b4d3e6f9ae4bed061ecab88f148e7d150f8fa279 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by zzzeek** [comment]
some pep8 stuff

**Comment by vrajmohan** [comment]
I'm new to `Gerrit`. Should I be squashing my commits so that I have only a _single_ commit in my PR?

**Comment by zzzeek** [comment]
> I'm new to `Gerrit`. Should I be squashing my commits so that I have only a _single_ commit in my PR?

I'm able to do the squashing from here, you can update the PR here in any way you see fit (new commits, squash, either way) and I can pull it over to the gerrit side automatically.

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision d2d8b92728c73b8c698cc6448e7e631e66f99d62 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset d2d8b92728c73b8c698cc6448e7e631e66f99d62 added to existing Gerrit review https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1094

File: examples/elementtree/pickle_type.py [hunk 1]
```diff
@@ -76,4 +77,7 @@ def __init__(self, name, element):
 document = session.query(Document).filter_by(filename="test.xml").first()
 
 # print
-document.element.write(sys.stdout)
+if py3k:
```
Comment (zzzeek) [inline examples/elementtree/pickle_type.py]: so, we need to make this change in optimized_ai and adjacency_list as well, right?  those still raise a "bytes" error.

Comment (vrajmohan) [reply examples/elementtree/pickle_type.py]: My fault, I was looking through the peephole from the issue. Addressing...

Comment (zzzeek) [reply examples/elementtree/pickle_type.py]: thanks for sticking to it !   :)

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision 8fed432b5d410ef197d1a28423fcdefade52f060 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset 8fed432b5d410ef197d1a28423fcdefade52f060 added to existing Gerrit review https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1094

**Review by sqla-tester — COMMENTED** [review]
OK, this is **sqla-tester** setting up my work to try to get revision ed90f3b77c2d0f4cc2cd54f40f5af29b489977d2 of this pull request into gerrit so we can run tests and reviews and stuff

**Comment by sqla-tester** [comment]
Patchset ed90f3b77c2d0f4cc2cd54f40f5af29b489977d2 added to existing Gerrit review https://gerrit.sqlalchemy.org/#/c/sqlalchemy/sqlalchemy/+/1094

**Comment by zzzeek** [comment]
I've been using [zimports](http://github.com/sqlalchemyorg/zimports) and it's working really well.

**Comment by vrajmohan** [comment]
I'm confused. Jenkins is complaining about a PEP8 issue _outside_ my changes?

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

> vrajmohan@github wrote:
 > 
 > I'm confused. Jenkins is complaining about a PEP8 issue _outside_
 > my changes?

me too. was it doing that originally or did that just pop up?

**Comment by sqla-tester** [comment]
**mike bayer** (zzzeek) wrote:

ah the pep8 failure is in master.   so if this got rebased against master that would pull the failure in.   now where did the rebase happen...in any case i can just fix it

**Pull request closed without merging**